### PR TITLE
Lock in profit on collaborative close price

### LINKED ISSUE #312: Profit not locked after collab settlement
Forgot to add the check for collab settlement in `Cfd::profit()`.

### PR DESCRIPTION
**Pull request by da-kami**
fixes #312 

Once we have collaborative close present in the state(s) we lock in the profit based on the price of the collaborative close.
Note: Currently no automated validation on the maker side if the given amounts and price of collaborative settlement are sane.

### CODE REVIEW

File: daemon/src/model/cfd.rs [hunk 1]
```diff
@@ -588,18 +590,19 @@ impl Cfd {
         let payout_curve =
             payout_curve::calculate(self.order.price, self.quantity_usd, self.order.leverage)?;
 
-        let current_price = current_price.try_into_u64()?;
+        let current_price_u64 = current_price.try_into_u64()?;
 
         let payout = payout_curve
             .iter()
-            .find(|&x| x.digits().range().contains(&current_price))
+            .find(|&x| x.digits().range().contains(&current_price_u64))
```
Comment (thomaseizinger) [inline daemon/src/model/cfd.rs]: Why is this rename necessary?

Comment (thomaseizinger) [reply daemon/src/model/cfd.rs]: Do everything in a new block if you don't want to shadow the original variable.

**Review by thomaseizinger — APPROVED** [review]
> Note: Currently no automated validation on the maker side if the given amounts and price of collaborative settlement are sane.

This needs a ticket.

**Comment by da-kami** [comment]
> > Note: Currently no automated validation on the maker side if the given amounts and price of collaborative settlement are sane.
> 
> This needs a ticket.

I felt this is more of a post MVP ticket, so recorded in the backlog here: https://github.com/comit-network/hermes/issues/319 

Please have a quick read through it @klochowicz and let me know if that makes sense from your point of view.

File: daemon/src/model/cfd.rs [hunk 2]
```diff
@@ -1280,15 +1284,15 @@ pub enum CfdStateChangeEvent {
...
+    ProposalSigned(CollaborativeSettlement),
     /// Maker signed and finalized the close transaction with both signatures
-    CloseSent(TimestampedTransaction),
+    CloseSent(CollaborativeSettlement),
 }
 
 /// Returns the Profit/Loss (P/L) as Bitcoin. Losses are capped by the provided margin
 fn calculate_profit(
     initial_price: Usd,
-    current_price: Usd,
```
Comment (klochowicz) [inline daemon/src/model/cfd.rs:1244]: isn't this function also used with the current price to present in the table when the Cfd is still Open? 

maybe start/end price?

**Review by klochowicz — APPROVED** [review]
nice one, thanks!

File: daemon/src/model/cfd.rs (hunk 2 above)
Comment (da-kami) [reply daemon/src/model/cfd.rs:1244]: not sure about the names, happy to rename in follow up 😅
